Libplanet.Net/Swarm.BlockSync.cs

```diff
@@ -466,34 +470,35 @@ in completedBlocks.WithCancellation(cancellationToken))
                 tipCandidate = tempTip;
                 _logger.Debug(
                     "TipCandidate: #{Index} {Hash}",
-                    tipCandidate?.Index,
-                    tipCandidate?.Hash);
+                    tipCandidate.Block?.Index,
+                    tipCandidate.Block?.Hash);
 
-                if (tipCandidate is null)
+                if (tipCandidate.Block is null)
                 {
                     // If there is no blocks in the network (or no consensus at least)
                     // it doesn't need to receive states from other peers at all.
                     return renderSwap;
                 }
 
-                var deltaBlocks = new LinkedList<Block<T>>();
+                var deltaBlocks = new LinkedList<(Block<T> Block, BlockCommit Commit)>();
                 while (true)
                 {
-                    Block<T> blockToAdd;
-                    if (deltaBlocks.First is LinkedListNode<Block<T>> node)
+                    (Block<T> Block, BlockCommit Commit) blockToAdd;
+                    if (deltaBlocks.First is LinkedListNode<(Block<T>, BlockCommit)> node)
                     {
-                        Block<T> b = node.Value;
+                        (Block<T> b, BlockCommit c) = node.Value;
                         if (b.PreviousHash is { } p && !workspace.ContainsBlock(p))
                         {
-                            blockToAdd = workspace.Store.GetBlock<T>(p);
+                            blockToAdd =
+                                (workspace.Store.GetBlock<T>(p), workspace.GetBlockCommit(p));
```

Comment:
This fails in some tests. It seems to sort the fetched block, so the previous `BlockCommit` is needed. 

Problem was throwing an exception in `GetBlockCommit()`. `GetBlockCommit()` works if the target `Block<T>` has been appended into the chain before, however, in this case, we have not yet appended a block.

```csharp
(Block<T> b, BlockCommit _) = node.Value; // <-- We don't BlockCommit of the current block.
if (b.PreviousHash is { } p && !workspace.ContainsBlock(p))
{
  Block<T> block = workspace.Store.GetBlock<T>(p);
  blockToAdd = (block, workspace.Store.GetBlockCommit(block.Index)); // <-- Take BlockCommit from store.
}
```

I think my simplest (not sure performance-wise) workaround is storing all `BlockCommit` and taking it from the store while doing it.

```csharp
_logger.Verbose(
    "Add a block #{BlockIndex} {BlockHash}...",
    block.Index,
    block.Hash
);
block.ValidateTimestamp();
workspace.Store.PutBlock(block);
workspace.Store.PutBlockCommit(commit); // <-- Store BlockCommit
if (tempTip.Block is null ||
    BlockChain.PerceiveBlock(block).Index >
        BlockChain.PerceiveBlock(tempTip.Block).Index)
{
    tempTip = (block, commit);
}
```